src/Storage/Commands.Storage/File/Cmdlet/RemoveAzureStorageShare.cs

```diff
@@ -77,9 +91,54 @@ public override void ExecuteCmdlet()
             {
                 this.RunTask(async taskId =>
                 {
+                    if (share.IsSnapshot && IncludeAllSnapshot.IsPresent)
+                    {
+                        throw new PSArgumentException(string.Format(CultureInfo.InvariantCulture, "'IncludeAllSnapshot' should only be specified to delete a base share, and should not be specified to delete a Share snapshot: {0}", share.SnapshotQualifiedUri));
+                    }
+
                     if (force || ShareIsEmpty(share) || ShouldContinue(string.Format("Remove share and all content in it: {0}", share.Name), ""))
                     {
-                        await this.Channel.DeleteShareAsync(share, null, this.RequestOptions, this.OperationContext, this.CmdletCancellationToken).ConfigureAwait(false);
+                        DeleteShareSnapshotsOption deleteShareSnapshotsOption = DeleteShareSnapshotsOption.None;
+                        bool retryDeleteSnapshot = false;
+
+                        //Force means will delete the share anyway, so use 'IncludeSnapshots' to delete the share even has snapshot, or delete will fail when share has snapshot
+                        // To delete a Share shapshot, must use 'None' 
+                        if (IncludeAllSnapshot.IsPresent)
+                        {
+                            deleteShareSnapshotsOption = DeleteShareSnapshotsOption.IncludeSnapshots;
+                        }
+                        else
+                        {
+                            retryDeleteSnapshot = true;
+                        }
+
+                        try
+                        {
+                            await this.Channel.DeleteShareAsync(share, deleteShareSnapshotsOption, null, this.RequestOptions, this.OperationContext, this.CmdletCancellationToken).ConfigureAwait(false);
+                            retryDeleteSnapshot = false;
+                        }
+                        catch (StorageException e)
+                        {
+                            //If x-ms-delete-snapshots is not specified on the request and the share has associated snapshots, the File service returns status code 409 (Conflict).
+                            if (!(e.IsConflictException() && retryDeleteSnapshot))
```

Comment:
Is there ever a case where DeleteShareAsync will throw a conflictexception when the deleteShareSnapshotsOption is set to IncludeSnapshots? I'm just curious about whether you need to check if retryDeleteSnapshot is true.

Reply:
There  are cases to throw 419 (conflict) error when the share is being deleted, and try to delete again.

So we need to check the retryDeleteSnapshot  is set and the error is 419 error. This the best we can do.

Reply:
Okay, sounds good.